Userland/Services/SpiceAgent/Message.cpp

```diff
@@ -206,4 +206,57 @@ ErrorOr<String> ClipboardMessage::debug_description()
     return builder.to_string();
 }
 
+ErrorOr<FileTransferStartMessage> FileTransferStartMessage::read_from_stream(AK::Stream& stream)
+{
+    auto id = TRY(stream.read_value<u32>());
+
+    auto metadata_bytes = TRY(stream.read_until_eof());
+    auto metadata_content = TRY(String::from_utf8(metadata_bytes));
+
+    // TODO: We need some sort of INIParser, or we need to make Core::ConfigFile not depend on having an actual Core::File
+    // The first line in the file should always be `[vdagent-file-xfer]`
+    auto lines = TRY(metadata_content.split('\n'));
+    if (lines.is_empty() || lines.at(0) != "[vdagent-file-xfer]") {
+        return Error::from_string_literal("Failed to parse file transfer metadata");
+    }
+
+    String name;
+    u32 size;
+
+    for (auto const& line : lines) {
+        // Ignore the header, we already assume that it is [vdagent-file-xfer]
+        if (line.starts_with('['))
+            continue;
+
+        if (line.starts_with_bytes("name="sv)) {
+            name = TRY(line.replace("name="sv, ""sv, ReplaceMode::FirstOnly));
+            if (name.is_empty()) {
+                return Error::from_string_literal("Failed to parse file name!");
+            }
+        }
+
+        if (line.starts_with_bytes("size="sv)) {
+            auto size_string = TRY(line.replace("size="sv, ""sv, ReplaceMode::FirstOnly));
+            auto optional_size = size_string.to_number<u32>(TrimWhitespace::Yes);
+            if (!optional_size.has_value()) {
+                return Error::from_string_literal("Failed to parse file size!");
+            }
+
+            size = optional_size.release_value();
+        }
+    }
```

Comment:
There's not any logic here to say that if we never got a name or a size, error "invalid msg from server". If you only send `[vdagent-file-xfer]\n`, this will make a FileTransferStartMessage with stack garbage as the size and an empty string name as written